MyToolsYourToolsBackend/MyToolsYourToolsBackend.Application/Services/UserGroupService.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using AutoMapper;
+using MyToolsYourToolsBackend.Application.Dtos;
+using MyToolsYourToolsBackend.Domain.DbContexts;
+using MyToolsYourToolsBackend.Domain.Entities;
+
+namespace MyToolsYourToolsBackend.Application.Services
+{
+    public class UserGroupService : IUserGroupService
+    {
+        private AppDbContext _dbContext;
+
+        public UserGroupService(AppDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+       
+        public UserGroupDto JoinGroup(UserGroupDto userGroup)
+        {
+            var userGroupToSave = Mapper.Map<UserGroup>(userGroup);
+
+            _dbContext.UserGroups.Add(userGroupToSave);
+
+            if (_dbContext.SaveChanges() == 0)
+            {
+                throw new Exception("Could not join the group");
+            }
+
+            return userGroup;
+        }
+
+        public bool LeaveGroup(UserGroupDto userGroup)
+        {
+            var userGroupToRemove = _dbContext.UserGroups.Find(new object[] { userGroup.UserId, userGroup.GroupId });
+
+            if (userGroupToRemove == null)
+            {
+                return false;
+            }
+
+            _dbContext.UserGroups.Remove(userGroupToRemove);
+
+            // check if it was last user in this group
+            var countOfRemainingGroupMembers = _dbContext.UserGroups.Where(ug => ug.GroupId == userGroup.GroupId).Count();
```

Comment:
Usuwanie grupy chyba mozna prosciej tak:
- var groupToRemove = _dbContext.Groups.Find(userGroup.GroupId);
if(!groupToRemove.UserGroups.Any(){
 _dbContext.Groups.Remove(groupToRemove);
}

Reply:
aa bo on moze miec jeszcze to jedna jedyną, no to w takim razie tak:
var groupToRemove = _dbContext.Groups.Find(userGroup.GroupId);
if(!groupToRemove.UserGroups.Any(ug=>ug.userId !=userGroup.UserId&& ug.groupId != userGroup.GroupId){
_dbContext.Groups.Remove(groupToRemove);
}

Reply:
no właśnie jak to testowałem to coś mi to tak średnio działało wiesz, bo generalnie to ja nwm czy te pola jak np. w encji User:
public ICollection<UserGroup> UserGroups { get;private set; } = new List<UserGroup>();
się uzupełniają automatycznie, jak się wstawi rekord do tabeli UserGroup?

Reply:
Po saveChanges powinno sie dodac